cmd/elegen/main.go

```diff
@@ -46,6 +58,18 @@ func main() {
 
 func generator(sets []spec.SpecificationSet, out string) error {
 
+	switch genType := viper.GetString("gen-type"); genType {
+	case "openapi3":
+		return genopenapi3.GeneratorFunc(sets, out)
+	case "", "elemental":
+		return genElemental(sets, out)
+	default:
+		return fmt.Errorf("unhandled generation type: '%s'", genType)
+	}
+}
+
+func genElemental(sets []spec.SpecificationSet, out string) error {
+
```

Comment:
the viper.Get should not be done sub functions. you should get all the viper flags once, and pass function parameters. this prevents creating a mess

Reply:
right... that is crossing logical systems boundaries.. will take care of it

Reply:
addressed [here](https://github.com/PaloAltoNetworks/elemental/pull/105/commits/4621c8907cbfecda3d957d810a9375d2c44c9d63)

Although that made the logic reversed, but it does respect scoping